### PR TITLE
DAOS-15615 test: Clear existing tmpfs mount points before running tests

### PR DESCRIPTION
**Pull request by phender**
Some test occassionaly fail to start servers due to insufficient available memory in CI due to left over DAOS mount points from a previous test.  Adding an option to launch.py to provided a filter, which if specified, will be used to umount and remove the directory for any mounted tmpfs filesystems matching the filter. When using --mode=ci the filter will be set to /mnt/daos.

Skip-unit-tests: true
Skip-fault-injection-test: true

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/launch_utils.py [hunk 1]
```diff
@@ -663,6 +675,117 @@ def _query_create_user(logger, hosts, user, gid=None, create=False):
...
+        Returns:
+            int: status code: 0 = success, 128 = failure
+        """
+        if not clear_mounts:
+            return 0
+
+        logger.debug("-" * 80)
+        hosts = test.host_info.all_hosts
+        logger.debug("Clearing existing mount points on %s: %s", hosts, clear_mounts)
+        command = f" df --type=tmpfs --output=target | grep -E '^({'|'.join(clear_mounts)})$'"
```
Comment (daltonbohning) [inline src/tests/ftest/util/launch_utils.py:695]:
```suggestion
        command = f"df --type=tmpfs --output=target | grep -E '^({'|'.join(clear_mounts)})$'"
```

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
All tests passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14295/13/testReport/

File: src/tests/ftest/util/launch_utils.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/util/launch_utils.py:695]: Ah, mind if I fix that in a follow-on?

**Comment by phender** [comment]
Example: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14295/13/artifact/Functional%20Hardware%20Large/launch/functional_hardware_large/job.log
- Before running the first test, we see no existing tmpfs mount points:
```
2024/05/17 09:53:32 DEBUG            _clear_mount_points: --------------------------------------------------------------------------------
2024/05/17 09:53:32 DEBUG            _clear_mount_points: Clearing existing mount points on wolf-[51,110-117]: ['/mnt/daos', '/mnt/daos0', '/mnt/daos1']
2024/05/17 09:53:32 DEBUG                     run_remote: Running on wolf-[51,110-117] with a 120 second timeout:  df --type=tmpfs --output=target | grep -E '^(/mnt/daos|/mnt/daos0|/mnt/daos1)$'
2024/05/17 09:53:32 DEBUG                log_result_data:   wolf-[51,110-117] (rc=1): <no output>
2024/05/17 09:53:32 DEBUG _remove_shared_memory_segments: Clearing existing shared memory segments on wolf-[51,110-117]
2024/05/17 09:53:32 DEBUG                     run_remote: Running on wolf-[51,110-117] with a 120 second timeout: ipcs -m
2024/05/17 09:53:32 DEBUG                log_result_data:   wolf-[51,110-117] (rc=0):
2024/05/17 09:53:32 DEBUG                log_result_data:     
2024/05/17 09:53:32 DEBUG                log_result_data:     ------ Shared Memory Segments --------
2024/05/17 09:53:32 DEBUG                log_result_data:     key        shmid      owner      perms      bytes      nattch     status      
2024/05/17 09:53:32 DEBUG                log_result_data:     
2024/05/17 09:53:32 DEBUG                _generate_certs: --------------------------------------------------------------------------------
```

- After a test that has run we then need to clean up the tmpfs mount points:
```
2024/05/17 09:59:14 DEBUG            _clear_mount_points: --------------------------------------------------------------------------------
2024/05/17 09:59:14 DEBUG            _clear_mount_points: Clearing existing mount points on wolf-[51,110-117]: ['/mnt/daos', '/mnt/daos0', '/mnt/daos1']
2024/05/17 09:59:14 DEBUG                     run_remote: Running on wolf-[51,110-117] with a 120 second timeout:  df --type=tmpfs --output=target | grep -E '^(/mnt/daos|/mnt/daos0|/mnt/daos1)$'
2024/05/17 09:59:14 DEBUG                log_result_data:   wolf-[51,110-117] (rc=1): <no output>
2024/05/17 09:59:14 DEBUG _remove_shared_memory_segments: Clearing existing shared memory segments on wolf-[51,110-117]
2024/05/17 09:59:14 DEBUG                     run_remote: Running on wolf-[51,110-117] with a 120 second timeout: ipcs -m
2024/05/17 09:59:15 DEBUG                log_result_data:   wolf-[113-114] (rc=0):
2024/05/17 09:59:15 DEBUG                log_result_data:     
2024/05/17 09:59:15 DEBUG                log_result_data:     ------ Shared Memory Segments --------
2024/05/17 09:59:15 DEBUG                log_result_data:     key        shmid      owner      perms      bytes      nattch     status      
2024/05/17 09:59:15 DEBUG                log_result_data:     0x10242048 12         daos_serve 660        1277952    0                       
2024/05/17 09:59:15 DEBUG                log_result_data:     0x10242049 13         daos_serve 660        1277952    0                       
2024/05/17 09:59:15 DEBUG                log_result_data:     
2024/05/17 09:59:15 DEBUG                log_result_data:   wolf-[110-112] (rc=0):
2024/05/17 09:59:15 DEBUG                log_result_data:     
2024/05/17 09:59:15 DEBUG                log_result_data:     ------ Shared Memory Segments --------
2024/05/17 09:59:15 DEBUG                log_result_data:     key        shmid      owner      perms      bytes      nattch     status      
2024/05/17 09:59:15 DEBUG                log_result_data:     0x10242049 12         daos_serve 660        1277952    0                       
2024/05/17 09:59:15 DEBUG                log_result_data:     0x10242048 13         daos_serve 660        1277952    0                       
2024/05/17 09:59:15 DEBUG                log_result_data:     
2024/05/17 09:59:15 DEBUG                log_result_data:   wolf-[51,115-117] (rc=0):
2024/05/17 09:59:15 DEBUG                log_result_data:     
2024/05/17 09:59:15 DEBUG                log_result_data:     ------ Shared Memory Segments --------
2024/05/17 09:59:15 DEBUG                log_result_data:     key        shmid      owner      perms      bytes      nattch     status      
2024/05/17 09:59:15 DEBUG                log_result_data:     
2024/05/17 09:59:15 DEBUG _remove_shared_memory_segments: Clearing shared memory segment 0x10242048 on wolf-[110-114]:
2024/05/17 09:59:15 DEBUG                     run_remote: Running on wolf-[110-114] with a 120 second timeout: sudo ipcrm -M 0x10242048
2024/05/17 09:59:15 DEBUG                log_result_data:   wolf-[110-114] (rc=0): <no output>
2024/05/17 09:59:15 DEBUG _remove_shared_memory_segments: Clearing shared memory segment 0x10242049 on wolf-[110-114]:
2024/05/17 09:59:15 DEBUG                     run_remote: Running on wolf-[110-114] with a 120 second timeout: sudo ipcrm -M 0x10242049
2024/05/17 09:59:15 DEBUG                log_result_data:   wolf-[110-114] (rc=0): <no output>
2024/05/17 09:59:15 DEBUG                _generate_certs: --------------------------------------------------------------------------------
```

**Review by tanabarr — APPROVED** [review]

**Pull request merged**